src/SessionSetup/SessionSetupDialog.cpp

```diff
@@ -122,33 +97,27 @@ SessionSetupDialog::SessionSetupDialog(SshConnectionArtifacts* ssh_connection_ar
   } else {
     state_machine_.setInitialState(&state_local_);
     ui_->localProfilingWidget->GetRadioButton()->setChecked(true);
+    ui_->processListWidget->SetNameToSelect(absl::GetFlag(FLAGS_process_name));
   }
+
+  qRegisterMetaType<std::vector<ProcessInfo>>("std::vector<orbit_grpc_protos::ProcessInfo>");
```

Comment:
@beckerhe 

This is needed to send the process list from `ConnectToLocalWidget::ProcessListUpdated` to `ProcessListWidget::UpdateList`. I put this here because the connect happens in `SessionSetupDialog`. I don't now if there is a better location. 

Similarly `Q_DECLARE_METATYPE(orbit_grpc_protos::ProcessInfo);` is currently added to `ProcessItemModel.h` and I am not sure if thats a good idea.

Reply:
I think these register calls should always into the constructor of the class that sends out a signal with a certain parameter type. You don't have to worry about "double registering". You can safely call `qRegisterMetaType` twice.

The same goes for the Q_DECLARE_METATYPE macro. I think it should into the header that defines the class with the signal that needs it, so it should probably go into `ConnectToLocalWidget.h`.

Reply:
`qRegisterMetaType` -> done

`Q_DECLARE_METATYPE` can only be done once, so I put it in ProcessListWidget.h